scripts/test-protein-prep.py

```diff
@@ -0,0 +1,136 @@
+)
+from covid_moonshot_ml.modeling import (
+    align_receptor,
+    prep_receptor,
+    du_to_complex,
+    mutate_residues,
+)
+from covid_moonshot_ml.datasets.utils import (
+    save_openeye_pdb,
+    add_seqres,
+    seqres_to_res_list,
+    load_openeye_pdb,
+)
+from openeye import oechem
+
+
+def get_args():
+    parser = argparse.ArgumentParser(description="")
+
+    ## Input arguments
+    parser.add_argument(
+        "-i",
+        "--input_prot",
+        required=True,
+        type=str,
+        help="Path to pdb file of protein to prep.",
+    )
+    parser.add_argument(
+        "-r",
+        "--ref_prot",
+        required=True,
+        type=str,
+        help="Path to reference pdb to align to.",
+    )
+    parser.add_argument(
+        "-o",
+        "--output_dir",
+        required=True,
+        type=str,
+        help="Path to output_dir.",
+    )
+    parser.add_argument(
+        "-l",
+        "--loop_db",
+        required=False,
+        type=str,
+        help="Path to loop database.",
+    )
+    parser.add_argument(
+        "-s",
+        "--seqres_yaml",
+        default=None,
+        type=str,
+        help="Path to yaml file of SEQRES.",
+    )
+
+    return parser.parse_args()
+
+
+def main():
+    args = get_args()
+    base_file_name = os.path.splitext(os.path.split(args.input_prot)[1])[0]
+    print(base_file_name)
+    out_name = os.path.join(args.output_dir, base_file_name)
+
+    if args.seqres_yaml:
+        ## first add standard seqres info
+
+        with open(args.seqres_yaml) as f:
+            seqres_dict = yaml.safe_load(f)
+        seqres = seqres_dict["SEQRES"]
+
+        ## Get a list of 3-letter codes for the sequence
+        res_list = seqres_to_res_list(seqres)
+
+        ## Generate a new pdb file with the SEQRES we want
+        seqres_pdb = f"{out_name}_01seqres.pdb"
+        add_seqres(args.input_prot, seqres_str=seqres, pdb_out=seqres_pdb)
+
+        ## Load in the pdb file as an OE object
+        seqres_prot = load_openeye_pdb(seqres_pdb)
+
+        ## Mutate the residues to match the residue list
+        initial_prot = mutate_residues(seqres_prot, res_list)
+        mutated_fn = f"{out_name}_02mutated.pdb"
+        save_openeye_pdb(initial_prot, mutated_fn)
+    else:
+        initial_prot = load_openeye_pdb(args.input_prot)
+
+    ## For each chain, align the receptor to the reference while keeping both chains.
+    for mobile_chain in ["A", "B"]:
+        print(f"Running on chain {mobile_chain}")
+        aligned_prot = align_receptor(
+            input_prot=initial_prot,
+            ref_prot=args.ref_prot,
+            dimer=True,
+            mobile_chain=mobile_chain,
+            ref_chain="A",
+        )
+        aligned_fn = f"{out_name}_03aligned_chain{mobile_chain}.pdb"
+        save_openeye_pdb(aligned_prot, aligned_fn)
+
+        ## Prep the receptor using various SPRUCE options
+        site_residue = "HIS:41: :A"
```

Comment:
obviously it would be better to do this in a way that isn't hard-coded!

If we don't solve this in this PR then we should make an issue for it.